crates/control/src/role_assignment.rs

```diff
@@ -355,7 +269,6 @@ impl RoleAssignment {
         } else {
             self.role = new_role;
         }
-        self.team_ball = team_ball;
 
         if let Some(game_controller_state) = context.filtered_game_controller_state {
```

Comment:
Is it correct that `last_time_player_was_penalized` is updated **after** it is used in this cycle?

Reply:
Done